sync/src/block/extension.rs

```diff
@@ -103,6 +118,45 @@ impl Extension {
         }
     }
 
+    fn initial_state(client: Arc<Client>, snapshot_target: Option<(H256, u64)>) -> State {
```

Comment:
How about making it a constructor of `State`?

Reply:
Actually there's a patch that moves this function to `State::initial` method. I'll move that patch to here.